osquery/tables/system/darwin/unified_log.mm

```diff
@@ -0,0 +1,213 @@
+    {"timestamp", "date"},
+    {"message", "composedMessage"},
+    {"storage", "storeCategory"},
+    {"activity", "activityIdentifier"},
+    {"process", "process"},
+    {"pid", "processIdentifier"},
+    {"sender", "sender"},
+    {"tid", "threadIdentifier"},
+    {"subsystem", "subsystem"},
+    {"category", "category"}};
+
+const std::map<std::string, bool> columnIsNumeric = {{"timestamp", false},
+                                                     {"message", false},
+                                                     {"storage", true},
+                                                     {"activity", true},
+                                                     {"process", false},
+                                                     {"pid", true},
+                                                     {"sender", false},
+                                                     {"tid", true},
+                                                     {"subsystem", false},
+                                                     {"category", false}};
+
+std::string convertLikeExpr(const std::string& value) {
+  // for the LIKE operator in NSPredicates, '*' matches 0 or more characters
+  //   and '?' matches a single character
+  std::string res = ba::replace_all_copy(value, "%", "*");
+  ba::replace_all(res, "_", "?");
+  return res;
+}
+
+void addQueryOp(NSMutableArray* preds,
+                const std::string& key,
+                const std::string& value,
+                ConstraintOperator op) {
+  if (supportedOps.count(op) > 0 && columnToOSLogEntryProp.count(key) > 0) {
+    std::string modified_val = value;
+    std::string modified_key = columnToOSLogEntryProp.at(key);
+    if (op == LIKE) {
+      modified_val = convertLikeExpr(value);
+    }
+    NSExpression* keyExp = [NSExpression
+        expressionForKeyPath:[NSString
+                                 stringWithUTF8String:modified_key.c_str()]];
+    NSString* valStr = [NSString stringWithUTF8String:modified_val.c_str()];
+
+    NSExpression* valExp = nil;
+    if (key == "timestamp") {
+      double provided_timestamp = [valStr doubleValue];
+      valExp = [NSExpression
+          expressionForConstantValue:
+              [NSDate dateWithTimeIntervalSince1970:provided_timestamp]];
+    } else if (columnIsNumeric.at(key)) {
+      valExp =
+          [NSExpression expressionWithFormat:@"%lld", [valStr longLongValue]];
+    } else {
+      valExp = [NSExpression expressionForConstantValue:valStr];
+    }
+
+    NSPredicate* pred = [NSComparisonPredicate
+        predicateWithLeftExpression:keyExp
+                    rightExpression:valExp
+                           modifier:NSDirectPredicateModifier
+                               type:supportedOps.at(op)
+                            options:0];
+    [preds addObject:pred];
+  }
+}
+
+QueryData genUnifiedLog(QueryContext& queryContext) {
+  QueryData results;
+  if (@available(macOS 10.15, *)) {
```

Comment:
This might be easier to read if you returned early when NSLog is unsupported.

```
if (!@available(macOS 10.15, *)) {
  VLOG(1) << "OSLog framework is not available";
  return {};
}
```